File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        output = list()
        for i in range(0, len(nums)):
            output.append(0)
        output[0] = 1
        for i in range(1, len(nums)):
            output[i] = output[i-1]*nums[i-1]
        postfix = 1
        for i in range(len(nums)-1, -1, -1):
            output[i] *= postfix
            postfix *= nums[i]

        return output

File: Data Structures & Algorithms/two-integer-sum/submission-1.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();
        int[] output = new int[2];
        
        for (int i = 0; i < nums.length; i++) {
            int num = nums[i];
            int diff = target - num;

            if (map.containsKey(num)) {
                int[] numbers = {map.get(num), i};
                return numbers;
            }

            map.put(diff, i);

        }

        return output;
    }
}
